docs/arch-overview.md

```diff
@@ -0,0 +1,82 @@
+# K8ssandraCluster Components
+K8ssandraCluster is the primary custom resource with which you will with interact when using K8ssandra Operator. This document provides a high level overview of the K8ssandraCluster custom resource. 
+
+The following diagram highlights the key objects that comprise a K8ssandraCluster. The K8ssandraCluster spec is shown with highlighted parts pointing to corresponding objects.
+
+![](img/arch-overview.png)
+
+# Kubernetes Clusters
+The various components that make up a K8ssandraCluster can be spread across multiple Kubernetes clusters. One of those clusters must be designated as the control plane cluster. The other clusters are data plane clusters. In the diagram we have three Kubernetes clusters - `control-plane`, `east`, and `west`. The latter two make up the data plane.
+
+The K8ssandraCluster object must be created in the control plane cluster.
+
+K8ssandra Operator is deployed in each cluster. The operator consists of several controllers. Some controllers, notably the K8ssandraCluster controller, only run in the control plane cluster. This is the main reason that the K8ssandraCluster object needs be created in the control plane cluster. 
```

Comment:
It might be useful to describe (maybe not here, but elsewhere perhaps) what the k8ssandra-operator does in the data-plane.  You mention that some controllers don't run, but what do, and what is it doing at that level.